File: php/src/Advertisement/UI/Http/PublishAdvertisementController.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\UI\Http;

use Demo\App\Advertisement\Application\Command\PublishAdvertisement\PublishAdvertisementCommand;
use Demo\App\Advertisement\Application\Command\PublishAdvertisement\PublishAdvertisementUseCase;
use Demo\App\framework\FrameworkRequest;
use Demo\App\framework\FrameworkResponse;

final class PublishAdvertisementController
{
    public function __construct(private PublishAdvertisementUseCase $useCase)
    {
    }

    public function request(FrameworkRequest $request): FrameworkResponse
    {
        $command = new PublishAdvertisementCommand(
            ($request->content())['id'],
            ($request->content())['description'],
            ($request->content())['password'],
        );

        $this->useCase->execute($command);

        return new FrameworkResponse(
            FrameworkResponse::STATUS_CREATED,
            []
        );
    }
}

File: kotlin/src/main/kotlin/advertisement/ui/http/PublishAdvertisementController.kt
package advertisement.ui.http

import advertisement.application.publishAdvertisement.PublishAdvertisementCommand
import advertisement.application.publishAdvertisement.PublishAdvertisementUseCase
import framework.FrameworkRequest
import framework.FrameworkResponse

class PublishAdvertisementController(private val useCase: PublishAdvertisementUseCase) {

    fun execute(request: FrameworkRequest): FrameworkResponse {
        useCase.execute(
            PublishAdvertisementCommand(
                request.content["id"]!!,
                request.content["description"]!!,
                request.content["password"]!!,
            )
        )

        return FrameworkResponse(
            FrameworkResponse.STATUS_CREATED,
            mapOf(),
        )
    }
}